api/v1alpha1/releaseadmissionplan_types.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"github.com/redhat-appstudio/release-service/kcp"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// ReleaseAdmissionPlanSpec defines the desired state of ReleaseAdmissionPlan.
+type ReleaseAdmissionPlanSpec struct {
+	// DisplayName is the long name of the ReleaseAdmissionPlan
+	// +optional
+	DisplayName string `json:"displayName"`
+
+	// Application is a reference to the application to be released in the managed workspace
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +required
+	Application string `json:"application"`
+
+	// Origin references where in the cluster to find the information required for a release
+	// +required
+	Origin kcp.NamespaceReference `json:"origin"`
+
+	// Environment defines which Environment will be used to release the application
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +required
```

Comment:
do we know that this is required? We don't do anything with it. Why have it be required already?

Reply:
It will be introduced just after this change, but maybe you are right. There could be cases in which we don't want a deployment I guess...